### ISSUE TITLE
bug: pagination seems to break the order of Items

### ISSUE THREAD

**Issue by brunocorrea23** [denoland/saaskit#231]
**Describe the bug**
Now the last item on the first page has zero upvotes.
The first item on the second page has two upvotes.
I guess this is happening because the ordering is being applied within the context of a page (ie: after the pagination split the items into pages). It should probably happen before.

**Expected behavior**
If a item on a second page has more upvotes than an item in the first page, it should be moved to the first page.

**Comment by iuioiua** [comment]
Ah, yes. Good point. Hm... I think a few things need to be done here:
1. Create a new `["items", currentYear, currentMonth, currentDate]` index for the homepage to use.
2. Get all items with the prefix `["items", currentYear, currentMonth]` and then order by votes.
3. Slice all results into portions of 10.
4. Have a `?page=` parameter that controls which portion (page) to show.

This will be a step towards #211.

**Comment by iuioiua** [comment]
Using `["items", currentYear, currentMonth, currentDate, score]` is also another solution worth considering if `kv.list()` orders and portions results in a desirable manner.

**Comment by huai-jie** [comment]
> 1. Create a new `["items", currentYear, currentMonth, currentDate]` index for the homepage to use.


Why not `["items", currentYear, currentMonth, currentDate, itemid]` ?

**Comment by iuioiua** [comment]
Oh, yes, I left out `itemId` - corrected.

**Comment by brunocorrea23** [comment]
If I understood correctly, this approach of segmenting by month first could have a weird behavior in some edge cases. Ex: in the beginning of the month, the homepage could be empty, or have only 2 items. 
It could be patched by running a first get, checking the results and running a second one, if necessary.

Ideally it would be better to first order the items by score and only then apply a filter. Not sure if it's currently possible to do that with kv

**Comment by iuioiua** [comment]
I've thought about this a little more. If we did `[items, Date.now()]` in `createItem()` and got entries by `kv.list({ prefix: [items], start: ["items", xMsAgo] })`, we'd get all items from x milliseconds ago. Also, I'd prefer not having `score` in the index as it's mutable and complicates `createVote()` and `deleteVote()` logic more than I'd prefer.

I'll open a PR for this and request your review, @brunocorrea23. Apologies that I didn't get back to you before you opened #236.

**Comment by brunocorrea23** [comment]
@iuioiua, no problem at all about my PR :)

I understand your concern about having the `score`  in the index. But while it's true that the `createVote()` and `deleteVote()` logic would become more complicated, the 'read' operation would be much simpler, since the ordering would already be done - it comes out of the box from kv.

About the solution you've just proposed: I'd love to take a look at your PR, thanks.
But not sure if I understood the idea. I understood you want to have the posts ordered by date in kv. The read operation would retrieve all itens within a certain rolling window of, say, 7 days. 
Some comments:
- not sure if you want to use the `{ limit: 10 }` in the `kv.list` function. If you do, I think you will run into the same problem as the current implementation, since there is no way to pre-order by score.
- if you choose to retrieve all elements within this time window, then there is no guarantee of how many items will be retrieved and the pagination benefits gets reduced (of course it is better than retrieving all items and then ordering). Iterating pages might become more difficult as well. 
- also,  if there is an item with a high vote count outside the original time window, I think it would probably be out of order.

In summary, I think that any solution that filters before ordering would have its edge cases.
